### PR TITLE
[js/webgpu] Change timestamp-query-in-passes to timestamp-query

### PR DESCRIPTION
**Pull request by gyagp**
Timestamp-query has a broader support than timestamp-query-in-passes on all the platforms, including macOS. 
Note that to enable timestamp-query, you still need to add switch "--enable-dawn-features=allow_unsafe_apis" to Chrome. By default, the lowest 16 bits are masked with 0 (at a granularity about 0.1ms) for privacy. To get the highest precision, you need to add another switch "--enable-webgpu-developer-features".

### CODE REVIEW

**Comment by gyagp** [comment]
run web CI

**Comment by gyagp** [comment]
@qjia7, please take a look.

File: js/web/lib/wasm/jsep/backend-webgpu.ts [hunk 1]
```diff
@@ -197,21 +195,13 @@ export class WebGpuBackend {
...
   }
 
   dispose(): void {
-    // currently, we do not do anything in this function. In all known use cases, we don't have the requirement to
-    // actually dispose the WebGpuBackend instance, because it's always used as a singleton.
-    //
-    // revisit this place if we get real requirement to dispose the instance.
+    if (typeof this.querySet !== 'undefined') {
+      this.querySet.destroy();
+    }
```
Comment (qjia7) [inline js/web/lib/wasm/jsep/backend-webgpu.ts:204]: nit: It's better to also add `this.gpuDataManager.dispose()` here.

**Review by qjia7 — APPROVED** [review]
LGTM, thanks.

**Comment by gyagp** [comment]
run web CI

**Comment by gyagp** [comment]
@guschmue @satyajandhyala @fs-eire Please review this change, thanks!

File: js/web/lib/wasm/jsep/backend-webgpu.ts (hunk 1 above)
Comment (gyagp) [reply js/web/lib/wasm/jsep/backend-webgpu.ts:204]: Done

**Review by guschmue — APPROVED** [review]

**Pull request merged**